libexec/bats-core/bats

```diff
@@ -84,94 +84,111 @@ expand_path() {
+    for exclude_file in "${EXCLUDE_FILES[@]}"; do
+      file_match=1
+      # 'file check' needed: $exclude_file can contain dir or file filter
+      if is_bats_file "$exclude_file"; then
+        if [[ "$suite_filename" == *"$exclude_file"* ]]; then
+          file_match=0
+          break
+        fi
       fi
+    done
+    if [[ "$file_match" -eq 1 ]]; then
+      filtered_suite_filenames+="${suite_filename} "
     fi
   done
 
-  return 1
+  printf -v "$result" '%s' "${filtered_suite_filenames%?}"
 }
 
-# argument: exclude_literal
+# arguments:
+#   exclude_pattern ('out' param)
+#   exclude_file 
 _build_exclude_pattern() {
-  local result="$2"
-  if is_bats_file "$1" ; then
-    printf -v "$result" '%s' "'*$1'" # file matching pattern
+  local result="$1"
+  local exclude_file="$2"
+
+  if is_bats_file "$exclude_file"; then
+    printf -v "$result" '%s' "'*$exclude_file'" # file matching pattern
   else
-    printf -v "$result" '%s' "'*/$1/*'" # directory matching pattern
+    printf -v "$result" '%s' "'*/$exclude_file/*'" # directory matching pattern
   fi
 }
 
-# argument: EXCLUDE_FILES
+# arguments:
+#   multi_exclude_pattern ('out' param)
+#   exclude_files 
 _build_multi_exclude_pattern() {
-  local result="$2"
-  # given performance concerns we concatenate and loop in a native c-style manner
-  local exclude_pattern
-  _build_exclude_pattern "${EXCLUDE_FILES[0]}" "excl_pattern"
-  # shellcheck disable=SC2154
-  exclude_pattern="-path $excl_pattern"
-  local files_length=${#EXCLUDE_FILES[@]}
-  for ((i=1;i<files_length;i++))
-  do
-    _build_exclude_pattern "${EXCLUDE_FILES[i]}" "excl_pattern"
-    exclude_pattern="${exclude_pattern} -o -path $excl_pattern"
+  local result="$1"
+  shift
+  local exclude_files=("$@")
+  
+  exclude_pattern=
+  _build_exclude_pattern 'exclude_pattern' "${exclude_files[0]}"
+
+  multi_exclude_pattern="-path $exclude_pattern"
+  local files_length="${#exclude_files[@]}"
+
+  for (( i=1; i<files_length; i++ )); do  
+    _build_exclude_pattern 'exclude_pattern' "${exclude_files[i]}"
+    multi_exclude_pattern+=" -o -path $exclude_pattern"
   done
-  printf -v "$result" %s "$exclude_pattern"
+  printf -v "$result" '%s' "$multi_exclude_pattern"
 }
 
-# builds and returns specific `find` command as string literal 
-# according given user input to the bat command.
+# dynamic building of parametrized `find` command as string 
+# literal according given user input to the bats command.
+# enables appropriate filtering!
 # 
-# internals: performance over fancy code
-# - using printf -v instead of command substitution for return values
-# - using find combined with -prune instead of piping with grep/awk etc.
-# - use of sub processes limited to the absolute minimum.
-#
-build_find_command_string() {
+# arguments:
+#   find_command ('out' param)
+#   exclude_files
+build_find_command() {
   local result="$1"
+  # shellcheck disable=SC2178
+  local exclude_files="$2"
+  
+    # shellcheck disable=SC2128
```

Comment:
disabled because i don't really agree with the warning. maybe i do miss something?!